### PR TITLE
APEXMALHAR-2316 Moved suitable code to activate() for initializing tu…

### PR DESCRIPTION
**Pull request by KapoorHitesh**
…ple class attribute. @bhupeshchawda @chinmaykolhatkar please review.

### CODE REVIEW

**Review by devtagare — COMMENTED** [review]
Changes seem fine.
Could you please check why the build failed ?

**Comment by KapoorHitesh** [comment]
@devtagare I analyzed the build failure and it appears to be failing because DocumentBuilder is using String argument instead of InputSource and hence the test application is throwing the same exception multiple times and the log limit is getting exceeded. 
Should I remove the test application and add it when I make the code changes for the above scenario?

**Comment by KapoorHitesh** [comment]
@devtagare have made the appropriate code changes please review

File: library/src/main/java/com/datatorrent/lib/parser/XmlParser.java [hunk 1]
```diff
@@ -35,14 +36,18 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.w3c.dom.Document;
+import org.xml.sax.InputSource;
 import org.xml.sax.SAXException;
+
```
Comment (bhupeshchawda) [inline library/src/main/java/com/datatorrent/lib/parser/XmlParser.java]: remove unnecessary changes

File: library/src/main/java/com/datatorrent/lib/parser/XmlParser.java [hunk 2]
```diff
@@ -94,9 +100,8 @@ public void processTuple(String inputTuple)
         DocumentBuilder builder;
         try {
           builder = factory.newDocumentBuilder();
-          Document doc = builder.parse(inputTuple);
+          Document doc = builder.parse(new InputSource(new ByteArrayInputStream(inputTuple.getBytes("utf-8"))));
           parsedOutput.emit(doc);
-
```
Comment (bhupeshchawda) [inline library/src/main/java/com/datatorrent/lib/parser/XmlParser.java:99]: unnecessary

File: library/src/main/java/com/datatorrent/lib/parser/XmlParser.java [hunk 3]
```diff
@@ -94,9 +100,8 @@ public void processTuple(String inputTuple)
         DocumentBuilder builder;
         try {
           builder = factory.newDocumentBuilder();
-          Document doc = builder.parse(inputTuple);
+          Document doc = builder.parse(new InputSource(new ByteArrayInputStream(inputTuple.getBytes("utf-8"))));
```
Comment (bhupeshchawda) [inline library/src/main/java/com/datatorrent/lib/parser/XmlParser.java]: Can you make it capital "UTF-8" ?

File: library/src/test/java/com/datatorrent/lib/parser/XmlParserApplicationTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+{
+  @Test
+  public void testApplication() throws IOException, Exception
+  {
+    try {
+      LocalMode lma = LocalMode.newInstance();
+      Configuration conf = new Configuration(false);
+      lma.prepareDAG(new XmlParserTest(), conf);
+      LocalMode.Controller lc = lma.getController();
+      lc.run(10000);// runs for 10 seconds and quits
```
Comment (bhupeshchawda) [inline library/src/test/java/com/datatorrent/lib/parser/XmlParserApplicationTest.java:67]: Not the right way to run a test. Need to validate the results after the test is run.
See https://github.com/apache/apex-malhar/blob/master/stream/src/test/java/org/apache/apex/malhar/stream/FunctionOperator/FunctionOperatorTest.java for an example.

File: library/src/test/java/com/datatorrent/lib/parser/XmlParserApplicationTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      public static boolean emitTuple = true;
+
+      public final transient DefaultOutputPort<String> output = new DefaultOutputPort<String>();
+
+      @Override
+      public void emitTuples()
+      {
+        if (emitTuple) {
+          output.emit(xmlSample);
+          emitTuple = false;
```
Comment (bhupeshchawda) [inline library/src/test/java/com/datatorrent/lib/parser/XmlParserApplicationTest.java]: Why emit just one tuple?

**Review by bhupeshchawda — CHANGES_REQUESTED** [review]

**Comment by KapoorHitesh** [comment]
@bhupeshchawda have made the suggested code changes. Please review.
